pkg/rbac/sharedData.go

```diff
@@ -138,6 +139,13 @@ func (shared *SharedData) PopulateSharedCache(ctx context.Context) {
 		klog.V(5).Info("Using shared data from cache.")
 		return
 	} else { // get data and add to cache
+
+		//create metric and set labels
+		HttpDurationByQuery := metric.HttpDurationByLabels(prometheus.Labels{"action": "create_shared_cache"})
+
+		//create timer and return observed duration
+		timer := prometheus.NewTimer(HttpDurationByQuery.WithLabelValues("GET", "200")) //change labels
```

Comment:
At this point we don't know that the status_code will be 200.
Most likely we'll need to add a new middleware that executes at the end of the request.

Reply:
we can get that from the request from authMiddleware, right?